File: LINKED LIST/SinglyLinear.c
#include<stdio.h>
#include<stdlib.h>
struct Node
{
    int data;
    struct Node * next;
};
typedef struct Node NODE;
typedef struct Node* PNODE;
typedef struct Node** PPNODE;
void Display(PNODE first)
{
    while (first!=NULL)
    {
       printf("| %d | ->",first->data);
       first=first->next;
    }
    printf("NULL");
}
int  Count(PNODE first)
{
    int iCount = 0;
    while(first!=NULL)
    {
        iCount++;
        first=first->next;
    }
    return iCount;
}
void InsertFirst(PPNODE first,int iNo)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = iNo;
    newn->next = NULL;

    if(*first==NULL)
    {
        *first=newn;
        newn->next=NULL;
    }
    else
    {
        newn->next = *first;
        *first = newn;
    }
}

void InsertLast(PPNODE first , int iNo)
{
    PNODE newn = (PNODE)malloc(sizeof(NODE));
    newn->data = iNo;
    newn->next = NULL;

    if(*first==NULL)
    {
        *first=newn;
        newn->next=NULL;
    }
    else
    {
        PNODE temp = *first;
        while (temp->next!=NULL)
        {
            temp= temp->next;
        }     
        temp->next = newn;
        newn->next=NULL;
    }
}

void InsertAtpos(PPNODE first,int iNo,int Pos)
{
    int iCount = 0;
    iCount = Count(*first);
    if(Pos<0 || Pos >iCount+1)
    {
       printf("invalid Input");
       return;
    }
    if(Pos==1)
    {
        InsertFirst(first,iNo);
    }
    else if(Pos==iCount+1)
    {
        InsertLast(first,iNo);
    }
    else
    {
        PNODE newn = NULL;
        newn = (PNODE)malloc(sizeof(NODE));
        newn->data=iNo;
        newn->next=NULL;
        PNODE temp = *first;
        for(int iCnt=1;iCnt<=Pos-2;iCnt++)
        {
            temp=temp->next;
        }
        newn->next=temp->next;
        temp->next=newn;
    }
}
void DeleteFirst(PPNODE first)
{
    if(*first==NULL)
    {
        return;
    }
    else 
    {
        PNODE temp = *first;
        *first=temp->next;
        free(temp);
    }
}
void DeleteLast(PPNODE first)
{
    if(*first==NULL)
    {
        return;
    }
    else 
    {
        PNODE temp = *first;
        while(temp->next->next!=NULL)
        {
            temp = temp -> next;
        }
        free(temp->next);
        temp->next=NULL;
    }

}
void DeleteAtPos(PPNODE first,int Pos)
{
    int iCount = 0;
    iCount = Count(*first);
    if(Pos<0 || Pos >iCount+1)
    {
       printf("invalid Input");
       return;
    }
    if(Pos==1)
    {
        DeleteFirst(first);
    }
    else if(Pos==iCount)
    {
        DeleteLast(first);
    }
    else
    {
        PNODE temp = *first;
        PNODE target = (*first)->next;
        for(int iCnt=1;iCnt<=Pos-2;iCnt++)
        {
            temp=temp->next;
            target=target->next;
        }
        temp->next=target->next;
        free(target);
    }
}
int main()
{
    PNODE Head =NULL;
    int iRet = 0;
    int iNo = 0;
    int iPos = 0;
    int ch = 0;
    do{
    printf("\n");
    printf("------------MENU------------------\n");
    printf("1.INSERT AT FIRST POSITION\n");
    printf("2.INSERT AT LAST POSITON\n");
    printf("3.INSERT AT ANY POSITION\n");
    printf("4.DELETE AT FIRST POSITION\n");
    printf("5.DELETE AT LAST POSITON\n");
    printf("6.DELETE AT ANY POSITION\n");
    printf("7.DISPLAY \n");
    printf("8.COUNT\n");
    printf("9.EXIT");
    printf("\n");

    printf("PLEASE ENTER YOUR CHOICE \n");
    scanf("%d",&ch);

    if(ch<1||ch>9)
    {
        printf("INVALID INPUT");
    }
    switch (ch)
    {
       case 1:
       printf("ENTER NUMBER YOU WANT TO INSERT:\n");
       scanf("%d",&iNo);
       InsertFirst(&Head,iNo);
       break;
       case 2:
       printf("ENTER NUMBER YOU WANT TO INSERT:\n");
       scanf("%d",&iNo);
       InsertLast(&Head,iNo);
       break;
       case 3:
       printf("ENTER NUMBER YOU WANT TO INSERT:\n");
       scanf("%d",&iNo);
       printf("ENTER POSITION :\n");
       scanf("%d",&iPos);
       InsertAtpos(&Head,iNo,iPos);
       break;
       case 4:
       DeleteFirst(&Head);
       break;
       case 5:
       DeleteLast(&Head);
       break;
       case 6:
       printf("ENTER POSITION TO DELETE :\n");
       scanf("%d",&iPos);
       DeleteAtPos(&Head,iPos);
       break;
       case 7:
       Display(Head);
       break;
       case 8:
       iRet=Count(Head);
       printf("NUMBER OF ELEMENTS IN LINKED LIST : %d",iRet);
       break;
       case 9:
       printf("EXISTING PROGRAM");

    }
    }while (ch!=9);
    return 0;
}

File: LINKED LIST/SinglyLinear.java
import java.util.Scanner;

class node {
    public int data;
    public node next;
}

class singly {
    public node first;
    public int iCount;

    public singly() {
        first = null;
        iCount = 0;
    }

    public void display() {
        if (first == null) {
            System.out.println("linked list is empty");
        }
        node temp = first;
        while (temp != null) {
            System.out.print("|" + temp.data + "|->");
            temp = temp.next;
        }
    }

    public void insertFirst(int no) {
        node newn = new node();
        newn.data = no;
        newn.next = null;
        if (first == null) {
            first = newn;
        } else {
            newn.next = first;
            first = newn;
        }
        iCount++;
    }

    public void insertLast(int no) {
        node newn = new node();
        newn.data = no;
        newn.next = null;
        if (first == null) {
            first = newn;
        } else {
            node temp = first;
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newn;
            newn.next = null;
        }
        iCount++;
    }

    public void insertAtpos(int no, int pos) {
        if (pos < 1 || pos > iCount + 1) {
            System.out.println("invalid position");
            return;
        }
        if (pos == 1) {
            insertFirst(no);
        } else if (pos == iCount + 1) {
            insertLast(no);
        } else {
            node newn = new node();
            newn.data = no;
            newn.next = null;
            node temp = first;
            for (int iCnt = 1; iCnt <= pos - 2; iCnt++) {
                temp = temp.next;
            }
            newn.next = temp.next;
            temp.next = newn;
            iCount++;
        }
    }

    public void deleteFirst() {
        if (first == null) {
            System.out.println("linked list is empty");
            return;
        } else if (first.next == null) {
            first = null;
        } else {
            first = first.next;
        }
        iCount--;
    }

    public void deleteLast() {
        if (first == null) {
            System.out.println("linked list is empty");
            return;
        } else if (first.next == null) {
            first = null;
        } else {
            node temp = first;
            while (temp.next.next != null) {
                temp = temp.next;
            }
            temp.next = null;
        }
        iCount--;
    }

    public void deleteAtpos(int pos) {
        if (pos < 1 || pos > iCount) {
            System.out.println("invalid position");
            return;
        }
        if (pos == 1) {
            deleteFirst();
        } else if (pos == iCount) {
            deleteLast();
        } else {
            node temp = first;
            for (int iCnt = 1; iCnt <= pos - 2; iCnt++) {
                temp = temp.next;
            }
            node target = temp.next;
            temp.next = target.next;
            iCount--;
        }
    }

    public int count() {
        return iCount;
    }
}

class singlyLL {
    public static void main(String args[]) {
        singly sobj = new singly();
        int iRet = 0;
        int iNo = 0;
        int iPos = 0;
        int ch = 0;
        Scanner s = new Scanner(System.in);
        do {
            System.out.println("------------MENU------------------");
            System.out.println("1.INSERT AT FIRST POSITION");
            System.out.println("2.INSERT AT LAST POSITON");
            System.out.println("3.INSERT AT ANY POSITION");
            System.out.println("4.DELETE AT FIRST POSITION");
            System.out.println("5.DELETE AT LAST POSITON");
            System.out.println("6.DELETE AT ANY POSITION");
            System.out.println("7.DISPLAY ");
            System.out.println("8.COUNT");
            System.out.println("9.EXIT");
            System.out.println("PLEASE ENTER YOUR CHOICE ");
            ch = s.nextInt();
            if (ch < 1 || ch > 9) {
                System.out.println("invalid input");
            }
            System.out.println();
            switch (ch) {

                case 1:
                    System.out.println("Enter Number you want to insert:");
                    iNo = s.nextInt();
                    sobj.insertFirst(iNo);
                    break;
                case 2:
                    System.out.println("Enter Number you want to insert:");
                    iNo = s.nextInt();
                    sobj.insertLast(iNo);
                    break;
                case 3:
                    System.out.println("Enter Number you want to insert:");
                    iNo = s.nextInt();
                    System.out.println("Enter position");
                    iPos = s.nextInt();
                    sobj.insertAtpos(iNo, iPos);
                    break;
                case 4:
                    sobj.deleteFirst();
                    break;
                case 5:
                    sobj.deleteLast();
                    break;
                case 6:
                    System.out.println("Enter position");
                    iPos = s.nextInt();
                    sobj.deleteAtpos(iPos);
                    break;
                case 7:
                    sobj.display();
                    System.out.println();
                    break;
                case 8:
                    iRet = sobj.count();
                    System.out.println("NUMBER OF ELEMENTS IN LINKED LIST :" + iRet);
                    break;
                case 9:
                    System.out.println("Existing program");
            }
        } while (ch != 9);

    }
}
